tuf/utils/role_sort.go

```diff
@@ -0,0 +1,28 @@
+package utils
+
+import (
+	"strings"
+)
+
+// RoleList is a list of roles
+type RoleList []string
+
+// Len returns the length of the list
+func (r RoleList) Len() int {
+	return len(r)
+}
+
+// Less returns true if the item at i should be sorted
```

Comment:
I agree that I think the algorithm does not require alphabetic ordering, and that segment ordering should be fine.  I think @diogomonica's point was that if we got it wrong, or we otherwise changed things later to introduce a bug which might be affected by the ordering, the tests wouldn't fail deterministically.

I'm fine with no alphabetic ordering, since it adds some extra computational penalty, but if so it might be good to actually test that deterministic ordering within a bucket doesn't matter - (e.g. if we can inject the sorting function in in a test, we can have it return a bunch of different sort orderings per bucket).